docs/assignment3.md

```diff
@@ -1 +1,306 @@
+- *void showScene(final Scene scene)*: Replaces the currently active Scene with another one  
+- *void showPane(final Class<? extends GamePane> pane)*: Replaces the current  **GamePane** with another  
+- *showPane(final Class<? extends GamePane> pane)*: Returns coresponding **GamePane** 
+- *SceneManager getInstance()*: returns the instance of the class
+
+
+## Object diagrams								
+Author(s): `Torstein Thomassen`
+
+This object diagram shows an example of objects with different states in our program.
+All the different attributes and variables has be fulled in with example data.
+
+This structure resembles a tree structure with branches and leafs. The tree structure is commonly used in many software design. It's an efficient and a good way to structure the relation between objects. 
+
+This creates a hierarchy where a parent can only directly affect the children. E.g the **Match** can't affect the **Player** 'directly', only the **Team** objects.       
+
+`Figure representing the UML object diagram`
+  
+![](imgs/ass3%20objDiagram.png)
+
+The diagram above shows an example over the whole *Main* class. Where there is create a *Game* object, and the *Game* object again creates a *league* object. 
+This is a league with two matches. Each match has two teams playing against each other, and each team consist of 11 players.  
+ 
+ 
+
+## State machine diagrams									
+Author(s): `Torstein Thomassen` & `Teona Moga`
+
+The state diagrams of the **Game** and **League** classes are shown below.
+
+`State diagram over League Class`
+![](imgs/LeagueState-A3.jpg)
+
+The state diagram above represents the different states a league can be in. When it is initialized, no rounds are played so the *winners* map is empty; the *initialTeams* attribute represents the league schedule and *leagueId* represents an unique id for the given league, both are instantiated in the constructor to the parameters passed by the **Game** class. **League** has one recursive method, called when the results of the created schedule, i.e. *initialTeams*, are requested. What breaks the recursion is having no more matches to play in a subsequent round, as there is only one team left (the winner of the league).
+
+
+`State diagram over Game Class`
+![](imgs/ass3%20stateDiagram%20Game.png)
+
+The state diagram above represents the state a Game can be in. It has three different properties that can be changes and accessed with different methods.   
+
+
+## Sequence diagrams									
+Author(s): `Teona Moga`, `Patricia Santana Vasquez`, `Noe Rousset`
+`Sequence diagram showing League creation and execution`
+![](imgs/teamcreation.png)
+
+This sequence diagram represent how the creation of the player's team occur and the interactions between the game and team editor.
+The player represent by the actor element provides a list of players for his team in the team editor. Once the "Save Team" button is pressed, createTeam() is called.
```

Comment:
User inputs players and then presses a button, one of these actions is abstracted away for no reason. you should model both